### ISSUE TITLE
'Esc' key unexpectedly propagates to parent nodes.

### ISSUE THREAD

**Issue by freshp86** [PolymerElements/iron-menu-behavior#77]
See repro at https://jsfiddle.net/no958fqr/.

This is causing problems, for example when a paper-dropdown-menu resides within a dialog. Hitting the 'esc' key when the dropdown menu is focused should only close the dropdown menu, but not the dialog. This is affecting Chrome's new Settings page, see https://bugs.chromium.org/p/chromium/issues/detail?id=638518.

/cc @danbeam

**Comment by danbeam** [comment]
@tjsavage

**Comment by JanMiksovsky** [comment]
@valdrinkoshi Is this already addressed by the fix for PolymerElements/iron-overlay-behavior#212?

**Comment by valdrinkoshi** [comment]
No, this is different, as the request is to have the overlay stop the event's propagation.
@freshp86 @danbeam `iron-overlay-behavior` already provides the event `iron-overlay-canceled` which detail is the source event. You could listen to that event and stop the propagation yourself too `event.detail.stopPropagation()`
